### DISCUSSION TITLE
SAML Federation Support

### DISCUSSION THREAD

**Discussion by glatzert** [zitadel/zitadel#8942] [Q&A]
Currently Zitadel seems to allow SAML Authentication flows (acting as SP) only in the form of registering an IDP and then Zitadel will add an SP EntityID (and Metadata Endpoint and such) for each entered IdP. Looking at this from an federation (e.g. eduGAIN, inCommon, DFN AAI, Switch AAI, etc. pp) use case, this is unfeasable to use. The metadata import alone will not work (e.g. using ~http://www.aai.dfn.de/metadata/dfn-aai-sp-metadata.xml~ http://www.aai.dfn.de/metadata/dfn-aai-idp-metadata.xml fails with SAML metadata is missing (INST-8vam1khq22)).

Also Zitadel would need to expose itself as a single EntityID for the whole federation and possibly distinguish the "organization" of each IDP via attributes instead of Metadata-Source.

Is there something planned to enable SAML federation support in Zitadel oder is this out-of-scope?

**Comment by glatzert** [comment]
To add some "requiements" or ideas here. To enable proper saml federation support Zitadel would need to:
- automatically update metadata from federations via the metadata url (this would also be good in non federation scenarios, since key-rollovers happen)
- allow creating a "saml federation identity provider" that acts as a single SAML service provider
- allow assigning idps from a federation to organizations
- allow enabling and disabling single federation partners
- have a where-are-you-from for the saml idps, possibly based on saml-metadata scope attribute or schacHomeOrganization

**Comment by fforootd** [comment]
Hm good question.

Would you like to use zitadel as sp or idp in these settings?

To my understanding the culprit here is that these trust networks have many idps and you only want to configure the trust to the network instead of each idp, right?

Tagging @livio-a for awarness

**Reply by glatzert** [reply]
> Would you like to use zitadel as sp or idp in these settings?

Often those things are called "federation gateways" or "proxy". It would act as _single_ SP in regard to the federation and as IDP locally for one or multiple applications.
Sometimes it's also called a "community AAI" (and there's even an implementation blueprint for those: https://aarc-community.org/final-aarc-bpa-community-first-approach/

But essentially your understanding is correct. Zitadel would need to act as a _single_ SP and have multiple partner IDPs, which are all defined in a federation metadata file as here: http://www.aai.dfn.de/metadata/dfn-aai-idp-metadata.xml (the link in my original discussion entry, was the one containing all service providers, but that's currently not my use-case - I just mixed up the links - sorry)

**Reply by fforootd** [reply]
Got it. Need to check how much an effort it is to change our current behaviour 😁 

CC @hifabienne
